pkg/services/object/acl/v2/util.go

```diff
@@ -216,17 +216,30 @@ func isOwnerFromKey(id *user.ID, key *keys.PublicKey) bool {
 	return id2.Equals(*id)
 }
 
-// isVerbCompatible checks that tokenVerb operation can create auxiliary op operation.
-func isVerbCompatible(tokenVerb, op eaclSDK.Operation) bool {
-	switch tokenVerb {
-	case eaclSDK.OperationGet:
-		return op == eaclSDK.OperationGet || op == eaclSDK.OperationHead
+// assertVerb checks that token verb corresponds to op.
+func assertVerb(tok sessionSDK.Object, op eaclSDK.Operation) bool {
+	//nolint:exhaustive
+	switch op {
+	case eaclSDK.OperationPut:
+		return tok.AssertVerb(sessionSDK.VerbObjectPut, sessionSDK.VerbObjectDelete)
 	case eaclSDK.OperationDelete:
-		return op == eaclSDK.OperationPut || op == eaclSDK.OperationHead ||
-			op == eaclSDK.OperationSearch
-	case eaclSDK.OperationRange, eaclSDK.OperationRangeHash:
-		return op == eaclSDK.OperationRange || op == eaclSDK.OperationHead
-	default:
-		return tokenVerb == op
-	}
+		return tok.AssertVerb(sessionSDK.VerbObjectDelete)
+	case eaclSDK.OperationGet:
+		return tok.AssertVerb(sessionSDK.VerbObjectGet)
+	case eaclSDK.OperationHead:
+		return tok.AssertVerb(
+			sessionSDK.VerbObjectHead,
+			sessionSDK.VerbObjectGet,
+			sessionSDK.VerbObjectDelete,
+			sessionSDK.VerbObjectRange,
+			sessionSDK.VerbObjectRangeHash)
+	case eaclSDK.OperationSearch:
+		return tok.AssertVerb(sessionSDK.VerbObjectSearch, sessionSDK.VerbObjectDelete)
+	case eaclSDK.OperationRange:
+		return tok.AssertVerb(sessionSDK.VerbObjectRange, sessionSDK.VerbObjectRangeHash)
+	case eaclSDK.OperationRangeHash:
+		return tok.AssertVerb(sessionSDK.VerbObjectRangeHash)
```

Comment:
Why did you change the semantics?

Reply:
Oh, sorry, you now switch by op, and previously the switch was done by the token verb.